File: RpnCalculator-Scala/src/rpn/scala/operators/SubtractionOperator.scala
package rpn.scala.operators

import rpn.scala.RpnStack

class SubtractionOperator extends Operator {
  val MINUS = "-"
    
  def performOperation(numbers: RpnStack): Int = {
    val rhs = numbers.pop
    val lhs = numbers.pop
    val result = lhs - rhs
    numbers.push(result)
    return result
  }

  def handlesOperatorCharacter(operand: String): Boolean = {
    operand == MINUS;
  }

}

File: src/main/java/rpn/operator/SubtractionOperator.java
package rpn.operator;

public class SubtractionOperator extends BinaryOperator {

	private static final String MINUS = "-";

	@Override
	protected int doOperation(int lhs, int rhs) {
		return lhs - rhs;
	}

	@Override
	public boolean handlesOperatorCharacter(String operand) {
		return MINUS.contentEquals(operand);
	}

}
